### PR TITLE
lease: fix typo and modify findExpiredLeases function

### PR DESCRIPTION
**Pull request by lorneli**
typo: implemetation  -> implementation

### CODE REVIEW

**Comment by xiang90** [comment]
lgtm. defer to @gyuho or @heyitsanthony 

thanks for the contribution.

**Comment by lorneli** [comment]
jenkins failure looks like irrelevant to lease.

```
=== RUN   TestV3KVInflightRangeRequests
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd7b100]

goroutine 119268 [running]:
github.com/coreos/bbolt.(*Tx).Bucket(0x0, 0x19ad35a, 0x3, 0x3, 0x1a289c0)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/bbolt/tx.go:101 +0x30
github.com/coreos/etcd/mvcc/backend.(*readTx).UnsafeRange(0xc422a0d360, 0x19ad35a, 0x3, 0x3, 0xc4258844a0, 0x11, 0x12, 0x0, 0x0, 0x0, ...)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/mvcc/backend/read_tx.go:75 +0x826
github.com/coreos/etcd/mvcc.(*storeTxnRead).rangeKeys(0xc422c7c7e0, 0xc42a1a6500, 0x12, 0x20, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, ...)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/mvcc/kvstore_txn.go:140 +0x826
github.com/coreos/etcd/mvcc.(*storeTxnRead).Range(0xc422c7c7e0, 0xc42a1a6500, 0x12, 0x20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4a2b00, ...)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/mvcc/kvstore_txn.go:45 +0xe3
github.com/coreos/etcd/mvcc.(*txnReadWrite).Range(0xc42091bab0, 0xc42a1a6500, 0x12, 0x20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x452600, ...)
	<autogenerated>:30 +0xf7
github.com/coreos/etcd/mvcc.(*metricsTxnWrite).Range(0xc422c7c810, 0xc42a1a6500, 0x12, 0x20, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4229e6100, ...)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/mvcc/metrics_txn.go:38 +0x12f
github.com/coreos/etcd/etcdserver.(*applierV3backend).Range(0xc42599c280, 0x19bd600, 0xc422c7c810, 0xc4205176c0, 0x0, 0x0, 0x0)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/apply.go:263 +0x323
github.com/coreos/etcd/etcdserver.(*EtcdServer).Range.func2()
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/v3_server.go:99 +0xa6
github.com/coreos/etcd/etcdserver.(*EtcdServer).doSerialize(0xc432988000, 0x7ff4d82371c0, 0xc422ed6300, 0xc42a1a6640, 0xc422407548, 0x5, 0x5bb819bddda23564)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/v3_server.go:531 +0x15a
github.com/coreos/etcd/etcdserver.(*EtcdServer).Range(0xc432988000, 0x7ff4d82371c0, 0xc422ed6300, 0xc4205176c0, 0x0, 0x12e0200, 0x48674c)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/v3_server.go:100 +0x25a
github.com/coreos/etcd/etcdserver/api/v3rpc.(*kvServer).Range(0xc42274d640, 0x7ff4d82371c0, 0xc422ed6300, 0xc4205176c0, 0x12db200, 0xc422407600, 0x19beb00)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/api/v3rpc/key.go:50 +0xbc
github.com/coreos/etcd/etcdserver/api/v3rpc.(*quotaKVServer).Range(0xc42274d6c0, 0x7ff4d82371c0, 0xc422ed6300, 0xc4205176c0, 0xc42274d6c0, 0x5, 0x13e9cbc)
	<autogenerated>:26 +0x87
github.com/coreos/etcd/etcdserver/etcdserverpb._KV_Range_Handler.func1(0x7ff4d82371c0, 0xc422ed6300, 0x13ab1c0, 0xc4205176c0, 0xc4259f8b90, 0xc422a949a0, 0x7, 0xc422a949a7)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:3308 +0xa2
github.com/grpc-ecosystem/go-grpc-prometheus.UnaryServerInterceptor(0x7ff4d82371c0, 0xc422ed6300, 0x13ab1c0, 0xc4205176c0, 0xc42a1a6540, 0xc42a1a65c0, 0xc420013000, 0x7ff4d83864b0, 0x0, 0x462484)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/grpc-ecosystem/go-grpc-prometheus/server.go:29 +0xfb
github.com/coreos/etcd/etcdserver/api/v3rpc.newUnaryInterceptor.func1(0x7ff4d82371c0, 0xc422ed6300, 0x13ab1c0, 0xc4205176c0, 0xc42a1a6540, 0xc42a1a65c0, 0x0, 0x16, 0xc420013000, 0x7ff4d83864b0)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/api/v3rpc/interceptor.go:57 +0x106
github.com/coreos/etcd/etcdserver/etcdserverpb._KV_Range_Handler(0x12db200, 0xc42274d6c0, 0x7ff4d82371c0, 0xc422ed6300, 0xc4259f8aa0, 0xc422c87500, 0x0, 0x0, 0x15, 0x0)
	/home/jenkins/workspace/etcd-proxy/gopath/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:3310 +0x1fc
google.golang.org/grpc.(*Server).processUnaryRPC(0xc422b69540, 0x19c1b60, 0xc4229dd340, 0xc42eb47600, 0xc429422d20, 0x19fbde0, 0xc422ed7980, 0x0, 0x0)
	/home/jenkins/workspace/etcd-proxy/gopath/src/google.golang.org/grpc/server.go:781 +0x1222
google.golang.org/grpc.(*Server).handleStream(0xc422b69540, 0x19c1b60, 0xc4229dd340, 0xc42eb47600, 0xc422ed7980)
	/home/jenkins/workspace/etcd-proxy/gopath/src/google.golang.org/grpc/server.go:981 +0x150f
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4294d44c0, 0xc422b69540, 0x19c1b60, 0xc4229dd340, 0xc42eb47600)
	/home/jenkins/workspace/etcd-proxy/gopath/src/google.golang.org/grpc/server.go:551 +0xb7
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/jenkins/workspace/etcd-proxy/gopath/src/google.golang.org/grpc/server.go:552 +0xb9
exit status 2
```

File: lease/lessor.go [hunk 1]
```diff
@@ -480,15 +480,13 @@ func (le *lessor) runLoop() {
 
 		le.mu.Lock()
 		if le.isPrimary() {
-			ls = le.findExpiredLeases()
+			// rate limit
+			revokeLimit := leaseRevokeRate / 2
```
Comment (gyuho) [inline lease/lessor.go]: Move lines 483,484 to line 477 (out of mutex)?

Comment (heyitsanthony) [reply lease/lessor.go]: or just have `ls = le.findExpiredLeases(leaseRevokeRate / 2)`

**Comment by lorneli** [comment]
Thanks for the review. Move `revokeLimit := leaseRevokeRate / 2` out of mutex. Keep `revokeLimit` variable for readability.

**Comment by heyitsanthony** [comment]
Please squash the typo+fakelessor patches into a `lease: fix typos` patch, then this should be good to merge. Thanks!

**Comment by lorneli** [comment]
squashed :smiley:

**Comment by xiang90** [comment]
CI failures are unrelated. merging.

**Pull request merged**